tests/_internal/test_find_files.py

```diff
@@ -2,118 +2,167 @@
 # Copyright (C) 2020 Arm Mbed. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 #
-from unittest import TestCase, mock
+import contextlib
 from pathlib import Path
-from pyfakefs.fake_filesystem_unittest import patchfs
+from tempfile import TemporaryDirectory
+from unittest import TestCase, mock
+from typing import Iterable
+
 from mbed_targets import MbedTargetBuildAttributes
 
-from mbed_build._internal.find_files import (
-    find_files,
-    exclude_using_mbedignore,
-    exclude_using_target_labels,
-    exclude_using_labels,
-)
-
-
-class TestFindFiles(TestCase):
-    @patchfs
-    def test_finds_files_by_name_in_given_root(self, fs):
-        files = [
-            Path("root", "folder_1", "file.txt"),
-            Path("root", "folder_1", "folder_1_1", "file.txt"),
-            Path("root", "folder_2", "file.txt"),
-        ]
-        for file in files:
-            fs.create_file(file)
+from mbed_build._internal.find_files import find_files, MbedignoreFilter, LabelFilter, BoardLabelFilter
 
-        subject = find_files("file.txt", directory=Path("root"))
 
-        self.assertEqual(list(subject), files)
+@contextlib.contextmanager
+def create_files(files: Iterable[Path]):
+    with TemporaryDirectory() as temp_directory:
```

Comment:
Have you found TemporaryDirectory to be faster than pyfakefs?

Reply:
I haven't done side by side comparison, but those are fast to run - i.e.: I don't see the `pyfakefs` induced wait'n'think on every `test_*` function.

Reply:
Heck, I shall check to see if I can easily compare the two. ;)

Reply:
So, I changed the context manager to:

```py
@contextlib.contextmanager
def create_files(files: Iterable[Path]):
    with Patcher():
        temp_directory = Path("swagdirectory")
        for file in files:
            file_path = temp_directory / file
            file_directory = file_path.parent
            file_directory.mkdir(parents=True, exist_ok=True)
            file_path.touch()
        yield temp_directory
```

- for the sake of testing I just put all the things in "swagdirectory"
- it required changing tests ever so slightly - `self.assertIn` calls needed to be nested under context manager, otherwise comparing paths doesn't work (very confused about this) - **this should not have an effect on performance**

---

Pyfakefs:
```
Executed in    2.07 secs   fish           external 
   usr time    1.91 secs  135.00 micros    1.91 secs 
   sys time    0.15 secs  554.00 micros    0.15 secs
```

TemporaryDirectory:

```
Executed in  696.47 millis    fish           external 
   usr time  560.32 millis  157.00 micros  560.17 millis 
   sys time  133.08 millis  531.00 micros  132.55 millis 
```

Reply:
Wow thats quite a difference!